### PR TITLE
Feature/start migrations synchronous

### PR DESCRIPTION
**Pull request by GhiocelAndrei**
Start Migrations synchronous before app start. Also check if migrations are run successfully or not. If they fail we don't start the app, will change it later.
Also, small change : Modified 'Token' to 'Secret' in appsettings.json and forgot to change it in Program.cs when adding Jwt Bearer.

### CODE REVIEW

File: IssueTrackerAPI/Program.cs [hunk 1]
```diff
@@ -49,7 +49,7 @@
         {
             ValidateIssuerSigningKey = true,
             IssuerSigningKey = new SymmetricSecurityKey(System.Text.Encoding.UTF8.GetBytes(
-                builder.Configuration.GetSection("AppSettings:Token").Value)),
+                builder.Configuration.GetSection("AppSettings:Secret").Value)),
```
Comment (cristi-paraschiv) [inline IssueTrackerAPI/Program.cs:52]: this change belongs to the other PR

File: IssueTrackerApplication/Services/MigrationService.cs [hunk 2]
```diff
@@ -1,33 +1,39 @@
...
+            .AddSqlServer()
+            .WithGlobalConnectionString(connectionString)
+            .ScanIn(typeof(IssueTracker.DataAccess.Migrations.AddRoleColumnToUsersTable).Assembly).For.All())
+            .AddLogging(config => config.AddFluentMigratorConsole())
+            .BuildServiceProvider(false);
 
-        public Task StartAsync(CancellationToken cancellationToken)
-        {
-            using (var scope = _scopeFactory.CreateScope())
+            using (var scope = serviceProvider.CreateScope())
```
Comment (cristi-paraschiv) [inline IssueTrackerApplication/Services/MigrationService.cs:22]: no need to nest the try/catch; you could have a using declaration
`using var scope = serviceProvider.CreateScope();`

File: IssueTrackerApplication/Services/MigrationService.cs [hunk 3]
```diff
@@ -1,33 +1,39 @@
...
             {
-                var runner = scope.ServiceProvider.GetRequiredService<IMigrationRunner>();
-                runner.MigrateUp();
+                try
+                {
+                    var runner = scope.ServiceProvider.GetRequiredService<IMigrationRunner>();
+                    runner.MigrateUp();
+                }
+                catch (Exception ex)
```
Comment (cristi-paraschiv) [inline IssueTrackerApplication/Services/MigrationService.cs:29]: what are we doing with the exception? you should just have a `catch` statement

**Comment by florincatrina** [comment]
Please rebase this PR, it's very hard to follow on actual changes

File: IssueTrackerApplication/Services/MigrationService.cs [hunk 4]
```diff
@@ -1,33 +1,39 @@
...
         {
-            _scopeFactory = scopeFactory;
-        }
+            var serviceProvider = services.AddFluentMigratorCore()
+            .ConfigureRunner(config => config
+            .AddSqlServer()
+            .WithGlobalConnectionString(connectionString)
+            .ScanIn(typeof(IssueTracker.DataAccess.Migrations.AddRoleColumnToUsersTable).Assembly).For.All())
+            .AddLogging(config => config.AddFluentMigratorConsole())
+            .BuildServiceProvider(false);
```
Comment (florincatrina) [inline IssueTrackerApplication/Services/MigrationService.cs:20]: You should avoid building the service provider again, you could use app.Services and obtain a IServiceScopeFactory to run the migrations (after this line: [IScopeFactory](https://github.com/GhiocelAndrei/IssueTrackerAPI/blob/cd2ec84da1116f37a8826d642b67d82756db42ed/IssueTrackerAPI/Program.cs#L60C11-L60C18))

File: IssueTrackerApplication/Services/MigrationService.cs [hunk 5]
```diff
@@ -1,33 +1,39 @@
...
-        private readonly IServiceScopeFactory _scopeFactory;
-
-        public MigrationService(IServiceScopeFactory scopeFactory)
+        public static bool StartMigration(this IServiceCollection services, string connectionString)
         {
-            _scopeFactory = scopeFactory;
-        }
+            var serviceProvider = services.AddFluentMigratorCore()
+            .ConfigureRunner(config => config
+            .AddSqlServer()
```
Comment (cristi-paraschiv) [inline IssueTrackerApplication/Services/MigrationService.cs:16]: indentation: is `.AddSqlServer` being called on `services` or on `config`?

File: IssueTrackerApplication/Services/MigrationService.cs (hunk 2 above)
Comment (GhiocelAndrei) [reply IssueTrackerApplication/Services/MigrationService.cs]: I solved this issue in the latest Commit.

File: IssueTrackerApplication/Services/MigrationService.cs (hunk 4 above)
Comment (GhiocelAndrei) [reply IssueTrackerApplication/Services/MigrationService.cs]: I solved this issue in the latest Commit.

File: IssueTrackerApplication/Services/MigrationService.cs (hunk 5 above)
Comment (GhiocelAndrei) [reply IssueTrackerApplication/Services/MigrationService.cs]: I solved this issue in the latest Commit.

File: IssueTrackerApplication/Services/MigrationService.cs (hunk 3 above)
Comment (GhiocelAndrei) [reply IssueTrackerApplication/Services/MigrationService.cs]: This Issue was solved in the latest Commit.

File: IssueTrackerApplication/Services/MigrationService.cs [hunk 6]
```diff
@@ -1,33 +1,24 @@
 ﻿using FluentMigrator.Runner;
```
Comment (cristi-paraschiv) [inline IssueTrackerApplication/Services/MigrationService.cs:1]: lots of unused imports
image

File: IssueTrackerAPI/Program.cs [hunk 7]
```diff
@@ -55,10 +56,23 @@
...
+try
+{
+    migratorRunner.MigrateUp();
+}
+catch (Exception ex)
+{
+    Console.WriteLine($"Migration failed with exception: {ex.Message}");
+    return;
+}
+
```
Comment (florincatrina) [inline IssueTrackerAPI/Program.cs:75]: This can be wrapped in an extension method to keep Program class clean (it gets really big really fast)

File: IssueTrackerApplication/Services/MigrationService.cs (hunk 6 above)
Comment (GhiocelAndrei) [reply IssueTrackerApplication/Services/MigrationService.cs:1]: Deleted unused imports in the latest Commit.

File: IssueTrackerAPI/Program.cs (hunk 7 above)
Comment (GhiocelAndrei) [reply IssueTrackerAPI/Program.cs]: Solved this issue in the latest commit.

**Review by cristi-paraschiv — APPROVED** [review]

**Review by florincatrina — APPROVED** [review]
